### PR TITLE
:bug:Source Braintree: Fix Validation Error in Transactions Stream

### LINKED ISSUE #15547: Source Braintree: connection fails to sync subscription and transaction streams
## Environment
- **Airbyte version**:  0.39.35-alpha
- **OS Version / Instance**: -
- **Deployment**: EKS
- **Source Connector and version**: Braintree 0.1.3
- **Destination Connector and version**: s3 0.3.13
- **Step where error happened**:Sync job 

## Current Behavior
- Some data are synced: `customer_stream`, `discount_stream`, `dispute_stream`, `merchant_account_stream`, `plan_stream`
- then it fails, when syncing `subscription_stream` and `transaction_stream`

## Expected Behavior
- Sync should work

## Logs

image

- I think these are the relevant logs for `subscription_stream`:
```
2022-08-11 07:33:51 [44msource[0m > Syncing stream: subscription_stream 
2022-08-11 07:33:54 [44msource[0m > Encountered an exception while reading stream SourceBraintree
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 108, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 141, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 185, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/airbyte/integration_code/source_braintree/streams.py", line 123, in read_records
    yield from self._collect_items(stream_slice)
  File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/airbyte/integration_code/source_braintree/streams.py", line 112, in _collect_items
    item = self.model(**item)
  File "pydantic/main.py", line 406, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 6 validation errors for Subscription
billing_period_start_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
first_billing_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
next_billing_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
paid_through_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
transactions -> 0 -> subscription_details -> billing_period_end_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
transactions -> 0 -> subscription_details -> billing_period_start_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
```

- and just now I saw the same for `transaction_stream` (not in the log attached below)
```
subscription_details -> billing_period_end_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
subscription_details -> billing_period_start_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 116, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 107, in run
    for message in generator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 112, in read
    raise e
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 108, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 141, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 185, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/airbyte/integration_code/source_braintree/streams.py", line 123, in read_records
    yield from self._collect_items(stream_slice)
  File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/airbyte/integration_code/source_braintree/streams.py", line 112, in _collect_items
    item = self.model(**item)
  File "pydantic/main.py", line 406, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 2 validation errors for Transaction
subscription_details -> billing_period_end_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
subscription_details -> billing_period_start_date
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
```

[logs-2239.txt](https://github.com/airbytehq/airbyte/files/9308482/logs-2239.txt)

## Steps to Reproduce
1. Set up source -> Braintree
Screenshot 2022-08-11 at 14 15 25

3. Set up destination -> s3
Screenshot 2022-08-11 at 14 14 07

4. Create connection
image

## Are you willing to submit a PR?
- if it is something I can easily fix?

### LINKED ISSUE #29692: Source Braintree:  pydantic.error_wrappers.ValidationError: 1 validation error for Transaction, custom_fields str type expected (type=type_error.str)
### Connector Name

destination-s3

### Connector Version

0.1.15

### What step the error happened?

None

### Revelant information

When sync is enabled an exception is thrown and the replication process is cancelled. Attached is the synchronization job log with the error.
[f460218f_41fb_4f15_8622_56e80d28d668_job_2_attempt_5_txt.txt](https://github.com/airbytehq/airbyte/files/12405655/f460218f_41fb_4f15_8622_56e80d28d668_job_2_attempt_5_txt.txt)

### Relevant log output

```shell
The excerpt from the log indicating the issue is as follow:

2023-08-22 06:23:37 source > 1 validation error for Transaction
custom_fields
  str type expected (type=type_error.str)
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module></module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 156, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 98, in run
    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.read(source_spec, config, config_catalog, state))
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 132, in read
    yield from self.source.read(self.logger, config, catalog, state)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 139, in read
    raise e
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 120, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 194, in _read_stream
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Fixes #29692
Fixes https://github.com/airbytehq/airbyte/issues/15547

## How
*Describe the solution*

## Recommended reading order
1. `transaction.py`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-braintree/source_braintree/schemas/transaction.py [hunk 1]
```diff
@@ -61,7 +63,7 @@ class Transaction(BaseModel):
     created_at: datetime
     credit_card_details: CreditCard
     currency_iso_code: str
-    custom_fields: str
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-braintree/source_braintree/schemas/transaction.py:64]: why we delete custom_fields here?

Comment (topefolorunso) [reply airbyte-integrations/connectors/source-braintree/source_braintree/schemas/transaction.py:64]: It's a custom object collection of field/value pairs, so it is dynamic - see here https://developer.paypal.com/braintree/docs/reference/response/transaction/python#custom_fields

But I made up for it by allowing additional fields here  - https://github.com/airbytehq/airbyte/pull/31489/files#diff-cce8c41640b4701bed6c37a533394d7a18f3636dc9cfbb401d4e7f87958db8b0R126

**Comment by rstml** [comment]
any change this PR can be prioritised? I'm having similar issue. I believe this may also fix #15547

**Review by marcosmarxm — APPROVED** [review]
Thanks @topefolorunso 

Confirmed the custom fields are working now.
image

**Pull request merged**